backtester/engine/setup_test.go

```diff
@@ -0,0 +1,369 @@
+	cfg.FundingSettings.ExchangeLevelFunding = []config.ExchangeLevelFunding{
+		{
+			ExchangeName: testExchange,
+			Asset:        asset.Spot,
+			Currency:     currency.BTC,
+			InitialFunds: leet,
+			TransferFee:  leet,
+		},
+		{
+			ExchangeName: testExchange,
+			Asset:        asset.Futures,
+			Currency:     currency.BTC,
+			InitialFunds: leet,
+			TransferFee:  leet,
+		},
+	}
+	_, err = NewFromConfig(cfg, "", "", false)
+	if !errors.Is(err, nil) {
+		t.Errorf("received: %v, expected: %v", err, nil)
+	}
+}
+
+func TestLoadDataAPI(t *testing.T) {
+	t.Parallel()
+	bt := BackTest{
+		Reports: &report.Data{},
+	}
+	cp := currency.NewPair(currency.BTC, currency.USDT)
+	cfg := &config.Config{
+		CurrencySettings: []config.CurrencySettings{
+			{
+				ExchangeName: "Binance",
+				Asset:        asset.Spot,
+				Base:         cp.Base,
+				Quote:        cp.Quote,
+				SpotDetails: &config.SpotDetails{
+					InitialQuoteFunds: &leet,
+				},
+				BuySide:  config.MinMax{},
+				SellSide: config.MinMax{},
+				MakerFee: &decimal.Zero,
+				TakerFee: &decimal.Zero,
+			},
+		},
+		DataSettings: config.DataSettings{
+			DataType: common.CandleStr,
+			Interval: gctkline.OneMin,
+			APIData: &config.APIData{
+				StartDate: time.Now().Add(-time.Minute * 5),
+				EndDate:   time.Now(),
+			}},
+		StrategySettings: config.StrategySettings{
+			Name: dollarcostaverage.Name,
+			CustomSettings: map[string]interface{}{
+				"hello": "moto",
+			},
+		},
+	}
+	em := engine.ExchangeManager{}
+	exch, err := em.NewExchangeByName("Binance")
+	if err != nil {
+		t.Fatal(err)
+	}
+	exch.SetDefaults()
+	b := exch.GetBase()
+	b.CurrencyPairs.Pairs = make(map[asset.Item]*currency.PairStore)
+	b.CurrencyPairs.Pairs[asset.Spot] = &currency.PairStore{
+		Available:     currency.Pairs{cp},
+		Enabled:       currency.Pairs{cp},
+		AssetEnabled:  convert.BoolPtr(true),
+		ConfigFormat:  &currency.PairFormat{Uppercase: true},
+		RequestFormat: &currency.PairFormat{Uppercase: true}}
+
+	_, err = bt.loadData(cfg, exch, cp, asset.Spot, false)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestLoadDataDatabase(t *testing.T) {
+	t.Parallel()
+	bt := BackTest{
+		Reports: &report.Data{},
+	}
+	cp := currency.NewPair(currency.BTC, currency.USDT)
+	cfg := &config.Config{
+		CurrencySettings: []config.CurrencySettings{
+			{
+				ExchangeName: "Binance",
+				Asset:        asset.Spot,
+				Base:         cp.Base,
+				Quote:        cp.Quote,
+				SpotDetails: &config.SpotDetails{
+					InitialQuoteFunds: &leet,
+				},
+				BuySide:  config.MinMax{},
+				SellSide: config.MinMax{},
+				MakerFee: &decimal.Zero,
+				TakerFee: &decimal.Zero,
+			},
+		},
+		DataSettings: config.DataSettings{
+			DataType: common.CandleStr,
+			Interval: gctkline.OneMin,
+			DatabaseData: &config.DatabaseData{
+				Config: database.Config{
+					Enabled: true,
+					Driver:  "sqlite3",
+					ConnectionDetails: drivers.ConnectionDetails{
+						Database: "gocryptotrader.db",
```

Comment:
This requires a tempdir + cleanup otherwise it will write a file to the main GCT dir if the file doesn't exist